userland/AppMakefile.mk

```diff
@@ -66,19 +100,19 @@ $$(BUILDDIR)/$(1):
 
 # First step doesn't actually compile, just generate header dependency information
 # More info on our approach here: http://stackoverflow.com/questions/97338
-$$(BUILDDIR)/$(1)/%.o: %.c | $$(BUILDDIR)/$(1)
+$$(BUILDDIR)/$(1)/%.o: %.c | $$(BUILDDIR)/$(1) $$(HDRS)
```

Comment:
No, this is akin to `C_SRCS`, it's a well-ish known variable where you can put headers that need to get built (more traditionally used for pre-compiled headers)